clusterloader2/pkg/flags/types.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flags
+
+import (
+	"strconv"
+
+	"github.com/spf13/pflag"
+)
+
+var _ pflag.Value = (*stringFlagFunc)(nil)
+var _ flagFunc = (*stringFlagFunc)(nil)
+var _ pflag.Value = (*intFlagFunc)(nil)
+var _ flagFunc = (*intFlagFunc)(nil)
+
+type flagFunc interface {
+	parseFunc() error
+}
+
+type stringFlagFunc struct {
+	v *string
+	f func() error
```

Comment:
Can we have a better name for it?
Maybe initialize? Or sth like that?

Reply:
Renamed:
- v -> valPtr
- f -> initializeFunc